nbextensions/bulkImportCell/bulkImportCell.js

```diff
@@ -377,42 +440,82 @@ define([
          * 1. if there's a tab showing, stop() it and detach it
          * 2. update the tabs state to be selected
          * @param {string} tab id of the tab to display
+         * @param {string} fileType id of the filetype we're swapping to
          */
-        function toggleTab(tab) {
+        function toggleTab(tab, fileType) {
+            // if we're toggling the currently selected tab off,
+            // then it should be turned off.
+            if (tab === state.tab.selected && tab !== null && !fileType) {
+                tab = null;
+            }
             state.tab.selected = tab;
+            stopWidget();
+
+            if (tab !== null) {
+                if (!fileType) {
+                    fileType = state.fileType.selected;
+                }
+                runTab(tab, fileType);
+            }
+            cellTabs.setState(state.tab);
+        }
+
+        function stopWidget() {
             if (tabWidget !== null) {
                 tabWidget.stop();
-                var widgetNode = ui.getElement('widget');
+                const widgetNode = ui.getElement('widget');
                 if (widgetNode.firstChild) {
                     widgetNode.removeChild(widgetNode.firstChild);
                 }
             }
+        }
 
+        /**
+         * Initializes a tab and runs its associated widget.
+         * This doesn't change any state, just runs what it's told to,
+         * and returns the widget's start() Promise.
+         * @param {string} tab
+         * @param {string} fileType
+         */
+        function runTab(tab, fileType) {
             tabWidget = tabSet.tabs[tab].widget.make({
                 bus: cellBus,
-                workspaceInfo: workspaceInfo,
-                cell: cell,
-                model: model,
-                spec: spec,
-                jobId: undefined,
+                cell,
+                model,
+                spec: specs[typesToFiles[state.fileType.selected].appId],
+                fileType,
+                jobId: undefined
```

Comment:
This update adds the correct spec and currently selected file type to the tab's `make` function.